db.go

```diff
@@ -878,6 +878,53 @@ func (db *DB) CleanTombstones() (err error) {
 	return errors.Wrap(db.reload(), "reload blocks")
 }
 
+// LabelNames returns all the unique label names present in the DB in sorted order.
+// Label names are filtered using the matchers give. The label name is considered as
+// the value for the matcher, hence the name of the matcher is ignored.
+func (db *DB) LabelNames() ([]string, error) {
+	labelNamesMap := make(map[string]struct{})
+
+	updateLabelNames := func(br BlockReader) error {
+		ir, err := br.Index()
+		if err != nil {
+			return errors.Wrap(err, "get IndexReader")
+		}
+		names, err := ir.LabelNames()
+		if err != nil {
+			return errors.Wrap(err, "LabelNames() from IndexReader")
+		}
+		for _, name := range names {
+			labelNamesMap[name] = struct{}{}
+		}
+		if err = ir.Close(); err != nil {
+			return errors.Wrap(err, "close IndexReader")
+		}
+		return nil
+	}
+
+	if err := updateLabelNames(db.head); err != nil {
+		return nil, err
+	}
+	db.mtx.RLock()
+	blocks := db.blocks[:]
+	db.mtx.RUnlock()
+	for _, b := range blocks {
+		if err := updateLabelNames(b); err != nil {
+			return nil, err
+		}
+	}
+
+	labelNames := make([]string, 0, len(labelNamesMap))
+	for name := range labelNamesMap {
+		labelNames = append(labelNames, name)
+	}
+	sort.Slice(labelNames, func(i, j int) bool {
+		return labelNames[i] < labelNames[j]
+	})
+
+	return labelNames, nil
```

Comment:
you can use direclty the strings sort func.

```suggestion
	return sort.Strings(labelNames), nil
```